WordPress/Tests/WP/CapitalPDangitUnitTest.inc

```diff
@@ -0,0 +1,173 @@
+<?php
+
+// Bad: This is a comment with WORDPRESS spelled incorrectly.
+
+/* Bad: This is a comment with wordpress spelled incorrectly. */
+
+/*
+ * Bad: This is a comment with Wordpress spelled incorrectly.
+ */
+
+/**
+ * Function comment
+ *
+ * @param string $wordpress OK: Here the incorrect spelling is OK to comply with the variable name rules.
+ */
+function something( $wordpress ) {} // OK.
+
+/**
+ * Function comment
+ *
+ * @param string $my_wordpress_test OK: Here the incorrect spelling is OK to comply with the variable name rules.
+ */
+function something( $my_wordpress_test ) {} // OK.
+
+/**
+ * Bad: In this comment wordPresss should be fixed.
+ *
+ * @param string $test Bad: In this comment word press should be fixed.
+ */
+function something( $test ) {} // OK.
+
+function wordpress_function() {} // OK - comply with the function name rules.
+
+class Wordpress_Something {} // Bad.
+class Something_Word_press_Something {} // Bad.
+class Something_wordpressss {} // Bad.
+class WordPress_Something {} // OK.
+class Something_WordPress {} // OK.
+
+echo 'This is an explanation about wordpress.'; // Bad.
+echo "This is an {$explanation} about wordpress."; // Bad.
+
+// Bad.
+echo <<<EOD
+This is an {$explanation} about wordpress.
+EOD;
+
+echo 'http://wordpress.org/something'; // OK - part of a URL.
+?>
+
+<div class="copyright"><?php printf( wp_kses_post( __( 'Powered by <a href="%s">WordPress</a>', 'theme-slug' ) ), 'https://wordpress.org/' ); ?></div><!-- OK. -->
+
+<p>Here we have an inline HTML tag with wordpress spelled incorrectly.</p><!-- Bad. -->
+<p>Here we have an inline HTML tag with WordPress spelled correctly.</p><!-- OK. -->
+
+<p>Here we have an inline HTML tag with a URL, of course this should be coded differently, but even when it isn't, it should be ignored, so here goes: http://wordpress.org/ spelled incorrectly.</p><!-- OK. -->
+
+<p>Here we have an inline HTML tag with wordpressers spelled incorrectly.</p><!-- OK, part of another word. -->
```

Comment:
The checking is current bound to complete words to prevent false positives, so `wordpress` as part of a larger word will not match.

I don't think we could blanket fix any larger words, though we could possible create a target list of common words which contain `wordpress` which could and should be fixed.

This target list will need some careful consideration, but suggestions of larger words which could be on there are welcome.